go-selinux/selinux_linux.go

```diff
@@ -329,7 +329,7 @@ func lSetFileLabel(fpath string, label string) error {
 			break
 		}
 		if err != unix.EINTR {
-			return &os.PathError{Op: "lsetxattr", Path: fpath, Err: err}
+			return &os.PathError{Op: fmt.Sprintf("lsetxattr(label=%q)", label), Path: fpath, Err: err}
```

Comment:
Not a hard blocker (and maybe it's why @kolyshkin used string-concatenation in his suggestion); I tend to avoid using `%q` in formatting more nowadays; it's a great option, but the downside is that the (double) quotes can become a bit noisy if these messages pass some layers, they will be escaped, for example, when passing over an API, or when logging, it may become something like;

```console
error="setxattr(label=\"system_u:object_r:bin_t:s0:c3,c4\") /some/path: invalid argument"
```

Assuming we don't expect any really horrible strings to be passed as label by someone, `%s` would probably still work as it would still be wrapped within the `(label=<some value>)`;

```console
error="setxattr(label=system_u:object_r:bin_t:s0:c3,c4) /some/path: invalid argument"
```

Probably less relevant for this module, but printing the raw value without quotes may also help in cases where (e.g.) a value _including_ quotes was passed (I had some cases where the user expected quotes to be stripped by their shell, but made a mistake).

Reply:
Modified